### PR TITLE
[enhancement] [jaxrs-spec] Add builders to models

### PR DESCRIPTION
**Pull request by shybovycha**
_Note: this is a resurrection of #4916, since that branch has been rebased on top of `master` and CIs did not quite like me changing the target branch_

As per #2415, adding builders generation for models.

The reasons for this change are:

1. currently, the setters do not have the set prefix, which makes models not comply with java beans requirements
2. constructing models becomes painful, specifically when they have numerous fields

Compare the current way to construct a model to the proposed one:

now:

```
new Model().property1(value1).property2(value2)

// getters/setters

myModel.getProperty1();
myModel.property1(value1);
```

proposed:

```
Model.builder().property1(value1).property2(value2).build()

// getters/setters

myModel.getProperty1();
myModel.setProperty1(value1);
```

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by wing328** [comment]
cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

File: modules/openapi-generator/src/main/resources/JavaJaxRS/spec/pojo.mustache [hunk 1]
```diff
@@ -1,52 +1,38 @@
-{{#useSwaggerAnnotations}}
-import io.swagger.annotations.*;
-{{/useSwaggerAnnotations}}
+{{#useSwaggerAnnotations}}import io.swagger.annotations.*;{{/useSwaggerAnnotations}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaJaxRS/spec/pojo.mustache]: Please revert this change to avoid a blank line when `useSwaggerAnnotations` is false.

**Comment by wing328** [comment]
I saw changes to the Dart, TypeScript petstore samples. What script(s) did you run to update the samples?

**Comment by shybovycha** [comment]
@wing328 I've just ran `./bin/utils/ensure-up-to-date --batch`, as in README, I believe

**Comment by wing328** [comment]
I've manually resolved the merge conflicts, updated with the latest master and update samples. All tests are good.
